mountpoint-s3/tests/fuse_tests/write_test.rs

```diff
@@ -448,3 +452,76 @@ fn out_of_order_write_test_s3(offset: i64) {
 fn out_of_order_write_test_mock(offset: i64) {
     out_of_order_write_test(crate::fuse_tests::mock_session::new, offset);
 }
+
+async fn write_with_storage_class_test<F>(creator_fn: F, storage_class: &str)
+where
+    F: FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    const KEY: &str = "new.txt";
+
+    let config = TestSessionConfig {
+        filesystem_config: S3FilesystemConfig {
+            storage_class: Some(storage_class.to_owned()),
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+    let (mount_point, _session, _test_client) = creator_fn("write_with_storage_class_test", config);
+
+    let path = mount_point.path().join(KEY);
+
+    write_file(path).unwrap();
+    let region = get_test_region();
+    let sdk_client = get_test_sdk_client(&region).await;
+    let (bucket, prefix) = get_test_bucket_and_prefix("write_with_storage_class_test");
+    let attributes = sdk_client
+        .get_object_attributes()
+        .bucket(bucket)
+        .key(KEY)
+        .object_attributes(aws_sdk_s3::model::ObjectAttributes::StorageClass)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(storage_class, attributes.storage_class.unwrap().as_str());
+}
+
+#[cfg(feature = "s3_tests")]
+#[test_case("INTELLIGENT_TIERING")]
+#[test_case("GLACIER")]
+fn write_with_storage_class_test_s3(storage_class: &str) {
```

Comment:
Do we need to add storage classes to the mock client to be able to test with that here?